sockeye/inference.py

```diff
@@ -639,8 +652,13 @@ def make_input_from_factored_string(sentence_id: int,
     factors = [[] for _ in range(model_num_source_factors - 1)]  # type: List[Tokens]
     for token_id, token in enumerate(data_io.get_tokens(factored_string)):
         pieces = token.split(delimiter)
-        utils.check_condition(all(pieces) and len(pieces) == model_num_source_factors,
-                              "Failed to parse %s factors from token '%s'" % (model_num_source_factors, token))
+
+        if not all(pieces) or len(pieces) != model_num_source_factors:
+            logger.error("Failed to parse %s factors at position %d ('%s') in '%s'" % (model_num_source_factors,
```

Comment:
the format string seems of. model_num_source_factors is an int, no? in the format string you have `%s`

Reply:
fixed